packages/cysync-core/src/pages/MainApp/Components/AssetAllocation.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  AllocationShare,
+  Table,
+  TableBody,
+  TableDataRow,
+  TableHeader,
+  TableHeaderData,
+  TableNameBox,
+  TableTitle,
+  Typography,
+} from '@cypherock/cysync-ui';
+import React, { useEffect, useState } from 'react';
+
+import { AssetIconNameBox } from '~/pages/MainApp/Components/AssetIconNameBox';
+import { CoinTypes } from '@cypherock/coins';
+
+type TableHeaderNames = 'Asset' | 'Price' | 'Amount' | 'Value' | 'Allocation';
+
+interface HeadersData {
+  comparator: (a: any, b: any) => number;
+  width?: string;
+  padding?: string;
+  $noFlex?: boolean;
+}
+
+interface AssetDataType {
+  name: CoinTypes;
+  symbol: string;
+  price: number;
+  amount: number;
+  value: number;
+  allocation: number;
+  color: string;
+}
+
+export const AssetAllocation = () => {
+  const [sortedBy, setSortedBy] = React.useState<TableHeaderNames>('Asset');
+  const [isAscending, setIsAscending] = useState(true);
+
+  const createComparator =
+    (key: keyof AssetDataType) => (a: AssetDataType, b: AssetDataType) => {
+      if (a[key] < b[key]) {
+        return isAscending ? -1 : 1;
+      }
+      if (a[key] > b[key]) {
+        return isAscending ? 1 : -1;
+      }
+      return 0;
+    };
+
+  const headersData: Record<TableHeaderNames, HeadersData> = {
+    Asset: {
+      padding: '16px 20px 16px 96px',
+      width: '300',
+      $noFlex: true,
+      comparator: createComparator('name'),
+    },
+    Price: {
+      comparator: createComparator('price'),
+    },
+    Amount: {
+      comparator: createComparator('amount'),
+    },
+    Value: {
+      comparator: createComparator('value'),
+    },
+    Allocation: {
+      width: '300',
+      $noFlex: true,
+      comparator: createComparator('allocation'),
+    },
+  };
+
+  const [data, setData] = useState<AssetDataType[]>([
+    {
+      name: 'Bitcoin',
+      symbol: 'BTC',
+      price: 16981.44,
+      amount: 0.0178,
+      value: 2.9827,
+      allocation: 23.55,
+      color: '#F89C2D',
+    },
+    {
+      name: 'Ethereum',
+      symbol: 'ETH',
+      price: 674.44,
+      amount: 0.0179,
+      value: 45.4527,
+      allocation: 74.47,
+      color: '#0085FF',
+    },
```

Comment:
This should have the `id` property, which will be used to display the icons. Currently, we are using `name` to display icon which maps to `CoinTypes` but they do not match.

Example: For bitcoin the data name is `Bitcoin` but in `CoinType` it's `bitcoin`